processor/samplingprocessor/probabilisticsamplerprocessor/probabilisticsampler.go

```diff
@@ -91,6 +107,30 @@ func (tsp *tracesamplerprocessor) Shutdown() error {
 	return nil
 }
 
+// parseSpanSamplingPriority checks if the span has the "sampling.priority" tag to
+// decide if the span should be sampled or not. The usage of the tag follows the
+// OpenTracing semantic tags:
+// https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table
+func parseSpanSamplingPriority(span *tracepb.Span) samplingPriority {
+	attribMap := span.GetAttributes().GetAttributeMap()
+	if attribMap == nil {
+		return deferDecision
+	}
+
+	samplingPriorityAttrib := attribMap["sampling.priority"]
```

Comment:
one suggestion will be to attempt to get it from `tracestate` as well. Going forward the best place for `sampling.priority` will be in tracestate so it can be automatically propagated and used by all downstream components

Reply:
Good suggestion, created an issue to track it https://github.com/open-telemetry/opentelemetry-collector/issues/476